### PR TITLE
Thermomachines can no longer work backwards.

### LINKED ISSUE #24419: Thermomachines set to heat can cool, and vice-versa
### Exploit Reports

- [X] I confirm this issue is not an exploit. (Required)

### BYOND Version

515.1623

### Issue Description

Thermomachines are able to heat gasses when they are set to cooling mode.
They are also able to cool gasses when set to heating mode.

### What did you expect to happen?

For the thermomachine to not heat up gasses when set to cool them and vice-versa.

### What happened instead?

The heating/cooling toggle literally has no effect other than the values you are allowed to enter.

### Why is this bad/What are the consequences?

Why does the mode toggle even exist if it doesn't actually change the behaviour of the machine? If the gas is colder/hotter than what your cooler/heater can make, it should just leave it alone instead of swapping to the opposite mode to force it back into line.

New engineers can make the SM do bad things if they use the stock thermomachines because it heats the space-cooled coolant up from 21K to 73K before it reaches the chamber.

### Steps to reproduce the issue.

Cool gas down. Set target value higher. Watch as the cooler heats.

Do the opposite with heating mode.

### When did the problem start happening?

Probably since forever.

### Extra information

_No response_

### Relevant log output/runtime error

_No response_

### PR DESCRIPTION
**Pull request by FunnyMan3595**
## What Does This PR Do
Thermomachines can no longer work backwards, heating while cooling, or cooling while heating.
Fixed #24419

## Why It's Good For The Game
Don't do the opposite of what you say.

## Testing
Heated a pipe,  lowered heating temp, pipe didn't cool.
Repeated the other way around.

## Changelog
:cl:
fix: Thermomachines can no longer work backwards, heating while cooling, or cooling while heating. You can actually turn them on in the SM setup without making things actively worse!
/:cl:

### CODE REVIEW

**Comment by YoureNotTheSharpest** [comment]
I can already tell this will be annoying if for if you accidentally make the oxygen can too cold and need to start over at recooling it

**Comment by DGamerL** [comment]
Honestly
Controversial opinion but we should just throw away the heating and cooling settings, and just have a target temperature

**Comment by Fordoxia** [comment]
> I can already tell this will be annoying if for if you accidentally make the oxygen can too cold and need to start over at recooling it

Have a thermomachine set to hot and another set to cold. 2 of the station distros are actually set up like that already.

**Comment by Fordoxia** [comment]
> Honestly Controversial opinion but we should just throw away the heating and cooling settings, and just have a target temperature

It *does* solve the noob trap of turning on the freezers in the SM (thus causing it to overheat because they're cooling down the space-cooled gas but in reverse, causing an engine fire).

File: code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm [hunk 1]
```diff
@@ -97,6 +97,13 @@
 	if(!on)
 		return 0
 
+	// Coolers don't heat.
+	if(air_contents.temperature <= target_temperature && cooling)
+		return 0
```
Comment (Contrabang) [inline code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm:102]: the return values mean nothing here, please remove them from your added code (and the surrounding code also if you would like)

**Review by Contrabang — DISMISSED** [review]

File: code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm (hunk 1 above)
Comment (FunnyMan3595) [reply code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm:102]: Of course they don't. Why would I expect existing code to be sensible in its returns?

**Review by S34NW — APPROVED** [review]

**Pull request merged**